File: webservice/Frikollection_Api/Frikollection_Api/DTOs/Collection/UserCollectionDto.cs
using Frikollection_Api.DTOs.Product;

namespace Frikollection_Api.DTOs.Collection
{
    public class UserCollectionDto
    {
        public Guid CollectionId { get; set; }

        public string Name { get; set; }

        public bool? Private { get; set; }

        public DateOnly? CreationDate { get; set; }

        public ICollection<ProductDto>? Products { get; set; } = new List<ProductDto>();
    }
}

File: mobile_desktop_app/FrikollectionMobileDesktop/composeApp/src/commonMain/kotlin/org/example/frikollection_mobile_desktop/models/collection/UserCollectionDto.kt
package org.example.frikollection_mobile_desktop.models.collection

import kotlinx.serialization.Serializable
import org.example.frikollection_mobile_desktop.models.product.ProductDto

@Serializable
data class UserCollectionDto(
    val collectionId: String,
    val name: String,
    val isPrivate: Boolean,
    val createdAt: String,
    val products: List<ProductDto>
)